modules/optimization/src/base/OptimizationApp.C

```diff
@@ -46,6 +44,14 @@ OptimizationApp::registerAll(Factory & f, ActionFactory & af, Syntax & syntax)
       "AddOptimizationReporterAction", "OptimizationReporter", "add_optimization_reporter");
   registerMooseObjectTask("add_optimization_reporter", OptimizationReporter, false);
   addTaskDependency("add_optimization_reporter", "add_reporter");
+
+  // General Optimization action
+  registerTask("auto_create_mesh", false);
```

Comment:
I'm not sure this will still work if you build an app that depends on both stochastic tools and optimization. Why do we need this new task anyways?

Reply:
I could just manually put in a mesh and null kernel and get rid of the action